### PR TITLE
Remove 67 lines of code by avoiding the 90s file io API

### PR DESCRIPTION
**Pull request by colourful-land**
By the way I think in the test cases we don't worry overwriting test output files - in fact avoiding overriting makes test ambiguous, since `gradle testClean && gradle test` is expected to write new contents instead of shying away from overwriting.

### CODE REVIEW

File: src/test/java/com/alphawallet/attestation/cheque/TestRedeemCheque.java [hunk 1]
```diff
@@ -71,25 +71,25 @@ public void testSunshine() {
     try {
       PublicKey pk;
       System.out.println("Signed attestation:");
-      System.out.println(DERUtility.printDER(attestedCheque.getAtt().getDerEncoding(), "SIGNEABLE"));
+      System.out.println(DERUtility.toPEM(attestedCheque.getAtt().getDerEncoding(), "SIGNEABLE"));
```
Comment (jot2re) [inline src/test/java/com/alphawallet/attestation/cheque/TestRedeemCheque.java:74]: The tests now only print out the object reference and not the actual PEM file in human readable base 64.

File: src/test/java/org/devcon/ticket/UseTicketTest.java [hunk 2]
```diff
@@ -68,26 +68,26 @@ public void testSunshine() {
     try {
       PublicKey pk;
       System.out.println("Signed attestation:");
-      System.out.println(DERUtility.printDER(attestedTicket.getAtt().getDerEncoding(), "SIGNABLE"));
+      System.out.println(DERUtility.toPEM(attestedTicket.getAtt().getDerEncoding(), "SIGNABLE"));
```
Comment (jot2re) [inline src/test/java/org/devcon/ticket/UseTicketTest.java:71]: The tests now only print out the object reference and not the actual PEM file in human readable base 64.

File: src/main/java/com/alphawallet/attestation/core/DERUtility.java [hunk 3]
```diff
@@ -104,30 +104,19 @@ public static BigInteger decodeSecret(byte[] secretBytes) {
...
       coder.encode(input, 0, input.length, outstream);
       byte[] encodedCert = outstream.toByteArray();
-      StringBuilder builder = new StringBuilder();
-      builder.append("-----BEGIN " + type + "-----\n");
-      addBytes(builder, encodedCert);
-      builder.append("-----END " + type + "-----");
-      return builder.toString();
+      ByteArrayOutputStream outputStream = new ByteArrayOutputStream( );
+      outputStream.write(("-----BEGIN " + type + "-----\n").getBytes(StandardCharsets.UTF_8));
+      outputStream.write(encodedCert);
```
Comment (jot2re) [inline src/main/java/com/alphawallet/attestation/core/DERUtility.java:115]: No new-line information gets encoded which is needed for valid PEM files the results in the command line demo not working anymore.

**Review by jot2re — CHANGES_REQUESTED** [review]
It is nice get get some lines of code removed by taking advance of newer API methods, but unfortunately the logic is changed a bit which result in issues with the command line Demo.

File: src/test/java/com/alphawallet/attestation/cheque/TestRedeemCheque.java (hunk 1 above)
Comment (colourful-land) [reply src/test/java/com/alphawallet/attestation/cheque/TestRedeemCheque.java]: Thanks fixed.

File: src/main/java/com/alphawallet/attestation/core/DERUtility.java (hunk 3 above)
Comment (colourful-land) [reply src/main/java/com/alphawallet/attestation/core/DERUtility.java]: new line added thanks.

**Review by jot2re — CHANGES_REQUESTED** [review]
It looks good and works now. However I still think that the linebreaks are strictly needed if we want to be PEM compliant. See https://tools.ietf.org/html/rfc1421#section-4.3.2.4 which says that the lines containing the encoding must be 64 characters long

**Comment by colourful-land** [comment]
> It looks good and works now. However I still think that the linebreaks are strictly needed if we want to be PEM compliant. See https://tools.ietf.org/html/rfc1421#section-4.3.2.4 which says that the lines containing the encoding must be 64 characters long

Oh, fixed plus findings:

1. bouncy castle didn't provide a line-wrapping encoder - had to use the one shipped with Java, strange because they should be more compatible with that rfc than Java's built in encoder.
2. the code at `main` branch actually wraps at 65th character instead of 64.

**Review by jot2re — APPROVED** [review]
Looks great now, thank!

**Pull request merged**